pkg/apis/postgres-operator.crunchydata.com/v1beta1/managed_postgrescluster_types.go

```diff
@@ -0,0 +1,204 @@
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum={aws,azure,gcp}
+	Provider string `json:"provider_id"`
+
+	// The provider region where the cluster is located.
+	// TODO(managedpostgrescluster): Would we want to allow all and have Bridge send an error?
+	// +kubebuilder:validation:Required
+	Region string `json:"region_id"`
+
+	// The name of the secret containing the API key and team id
+	// +kubebuilder:validation:Required
+	Secret string `json:"secret,omitempty"`
+
+	// The amount of storage available to the cluster in gigabytes.
+	// The amount must be an integer, followed by Gi (gibibytes) or G (gigabytes) to match Kubernetes conventions.
+	// If the amount is given in Gi, we round to the nearest G value.
+	// The minimum value allowed by Bridge is 10 GB.
+	// The maximum value allowed by Bridge is 65535 GB.
+	// +kubebuilder:validation:Required
+	Storage resource.Quantity `json:"storage"`
+}
+
+// ManagedPostgresClusterStatus defines the observed state of ManagedPostgresCluster
+type ManagedPostgresClusterStatus struct {
+	// The ID of the postgrescluster in Bridge, provided by Bridge API and null until then.
+	// +optional
+	ID string `json:"id,omitempty"`
+
+	// observedGeneration represents the .metadata.generation on which the status was based.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// conditions represent the observations of postgrescluster's current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors={"urn:alm:descriptor:io.kubernetes.conditions"}
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// The cluster as represented by Bridge
+	// +optional
+	Cluster *ClusterDetails `json:"clusterResponse,omitempty"`
+
+	// The cluster upgrade as represented by Bridge
+	// +optional
+	ClusterUpgrade *ClusterUpgrade `json:"clusterUpgradeResponse,omitempty"`
+}
+
+// Right now used for cluster create requests and cluster get responses
+type ClusterDetails struct {
+	ID              string             `json:"id,omitempty"`
+	IsHA            bool               `json:"is_ha,omitempty"`
+	Name            string             `json:"name,omitempty"`
+	Plan            string             `json:"plan_id,omitempty"`
+	MajorVersion    int                `json:"major_version,omitempty"`
+	PostgresVersion intstr.IntOrString `json:"postgres_version_id,omitempty"`
+	Provider        string             `json:"provider_id,omitempty"`
+	Region          string             `json:"region_id,omitempty"`
+	Storage         int64              `json:"storage,omitempty"`
+	Team            string             `json:"team_id,omitempty"`
+	State           string             `json:"state,omitempty"`
+	// TODO(managedpostgrescluster): add other fields, DiskUsage, Host, IsProtected, IsSuspended, CPU, Memory, etc.
+}
```

Comment:
Does this belong here? :thinking: Feels more like an intermediary struct we use for interacting with the Bridge API rather than part of the `managedpostgrescluster` API...

Reply:
We've got a ticket, so resolving